packages/eas-cli/src/utils/fingerprintCli.ts

```diff
@@ -15,22 +18,44 @@ export async function createFingerprintAsync(
   sources: object[];
   isDebugSource: boolean;
 } | null> {
-  // @expo/fingerprint is exported in the expo package for SDK 52+
-  const fingerprintPath = silentResolveFrom(projectDir, 'expo/fingerprint');
-  if (!fingerprintPath) {
+  if (process.env.EAS_SKIP_AUTO_FINGERPRINT) {
+    Log.log('Skipping project fingerprint');
     return null;
   }
-  const Fingerprint = require(fingerprintPath);
-  const fingerprintOptions: Record<string, any> = {};
-  if (options.platform) {
-    fingerprintOptions.platforms = [options.platform];
-  }
-  if (options.workflow === Workflow.MANAGED) {
-    fingerprintOptions.ignorePaths = ['android/**/*', 'ios/**/*'];
-  }
-  if (options.debug) {
-    fingerprintOptions.debug = true;
+
+  const timeoutId = setTimeout(() => {
+    Log.log('⌛️ Computing project fingerprint taking longer than expected...');
+    Log.log('⏩ To skip this step, set the environment variable: EAS_SKIP_AUTO_FINGERPRINT=1');
+  }, 5000);
+
+  const spinner = ora(`Computing project fingerprint`).start();
+  try {
+    // @expo/fingerprint is exported in the expo package for SDK 52+
+    const fingerprintPath = silentResolveFrom(projectDir, 'expo/fingerprint');
+    if (!fingerprintPath) {
+      return null;
+    }
```

Comment:
I put this bit of code before the spinner. I've also cleaned up the spinner in `finally` for the future in case we decide to add `return` before the spinner succeeds